placeSpecificContent/hq/courtyard/controlPanel/controlPanel_server.js

```diff
@@ -0,0 +1,406 @@
+        beneath the stage. If it is already very close to its target position, slam it into exact position and 
+        always ensure it is aligned correctly on the x and z axes. Use its distance to its target positon to determine 
+        the velocity it needs to get there. */ 
+        setUpMovement: function(entityID, targetPosition) {
+            var currentPosition = Entities.getEntityProperties(entityID, 'position').position;
+    
+            // This part will ensure that the observer sees movement immediately
+            if (entityID === roundtable) {
+                var courtyardProperties = Entities.getEntityProperties(courtyard, ['dimensions', 'position']);
+                var courtyardHeight = courtyardProperties.dimensions.y;
+                
+                // if the top of the table is below the top of the courtyard, reposition it to be just the top of 
+                // the courtyard. Using the top of the courtyard gives a little leeway to ensure the table ends up below it
+                if (currentPosition.y + tableHeight * HALF <= courtyardProperties.position.y + courtyardHeight * HALF) {
+                    currentPosition = {
+                        x: currentPosition.x,
+                        y: courtyardProperties.position.y - tableHeight * HALF,
+                        z: currentPosition.z
+                    };
+                    Entities.editEntity(roundtable, { position: currentPosition });
+                }
+            }
+            
+            var distanceToTarget = currentPosition.y.toFixed(FIXED_DECIMAL_PLACES) - 
+                targetPosition.y.toFixed(FIXED_DECIMAL_PLACES);
+            if (distanceToTarget.toFixed(FIXED_DECIMAL_PLACES) === 0 || 
+                Math.abs(distanceToTarget.toFixed(FIXED_DECIMAL_PLACES)) < TARGET_DEVIATION_ALLOWANCE) {
+                Entities.editEntity(entityID, {
+                    position: targetPosition
+                });
+                if (entityID === roundtable) {
+                    roundtableInPlace = true;
+                } else {
+                    courtyardInPlace = true;
+                }
+                return;
+            }
+            
+            // If the entity has somehow been moved far away, slam it into lowered position to avoid hitting anyone
+            if (Math.abs(distanceToTarget) > MAX_DISTANCE_TO_TRANSITION_ENTITIES_M && 
+                Math.abs(distanceToTarget) < -MAX_DISTANCE_TO_TRANSITION_ENTITIES_M) {
+                currentPosition = entityID === roundtable ? roundtableLoweredPosition : courtyardLoweredPosition;
+                Entities.editEntity(entityID, {
+                    position: currentPosition
+                });
+            // If entity is very close to target position, slam into exact position
+            } else if (Math.abs(distanceToTarget) < MIN_DISTANCE_TO_TRANSITION_ENTITIES_M && 
+                Math.abs(distanceToTarget) > -MIN_DISTANCE_TO_TRANSITION_ENTITIES_M) {
```

Comment:
this will never be true, right? `Math.abs` will always return a positive number. why do you have to do `Math.abs(distanceToTarget)` ? why isn't `distanceToTarget` always positive, since it's a distance?

Reply:
Explained above.